### ISSUE TITLE
Question about D loss

### ISSUE THREAD

**Issue by NNsauce** [eriklindernoren/Keras-GAN#193]
I am training a GANomaly based on DCGAN. My D's goal is just the same as most of yours: fake_img's output is fake and real_img's output is real. 
But what makes me crazy is that my D keeps thinking the real image is real and fake image is also real during my training, even just after the D is built.  I mean why a random weighted D's output is not random, but prefers to predict a real result. And why my training can not change my D's preference, it just does not work at all

here is my training:
            self.x, self.x_defect = self.get_data()
            # Select a random batch of images and encode/decode/encode
            self.discriminator.trainable = True
            z = self.encoder1.predict(self.x_defect)
            self.x_fake = self.generator.predict(z)
            # Train the discriminator (imgs are real, imgs_ are fake)
            # d_loss_real = self.discriminator.train_on_batch(self.x, real)
            d_loss_real = [0, 0]
            d_loss_fake = self.discriminator.train_on_batch(self.x_fake, fake)
            d_loss = 0.5 * np.add(d_loss_real, d_loss_fake)
            # Train the generator (z -> img is valid and img -> z is is invalid)
            # self.discriminator.trainable = False
            g_loss = self.ganomaly_model.train_on_batch(self.x_defect, [real, self.x])

who can help me?  PLEASE!!

**Comment by NNsauce** [comment]
I am a new GAN trainer......maybe I made a wrong G or D network,,,,,
